tests/test_recipe_parser.py

```diff
@@ -0,0 +1,239 @@
+import io
+import pytest
+
+from conda_forge_tick.recipe_parser._parser import (
+    _parse_jinja2_variables,
+    _munge_line,
+    _unmunge_line,
+    _demunge_jinja2_vars,
+    _remunge_jinja2_vars,
+    _replace_jinja2_vars,
+)
+
+from conda_forge_tick.recipe_parser import CondaMetaYAML, CONDA_SELECTOR
+
+
+def test_parsing():
+    meta_yaml = """\
+{% set name = 'val1' %}  # [py2k]
+{% set name = 'val2' %}#[py3k and win]
+{% set version = '4.5.6' %}
+
+package:
+  name: {{ name|lower }}
+
+source:
+  url: foobar
+  sha256: 1  # [py2k]
+  sha256: 5#[py3k and win]
+
+build:
+  number: 10
+"""
+
+    cm = CondaMetaYAML(meta_yaml)
+
+    # check the jinja2 keys
+    assert cm.jinja2_vars['name__###conda-selector###__py2k'] == 'val1'
+    assert cm.jinja2_vars['name__###conda-selector###__py3k and win'] == 'val2'
+    assert cm.jinja2_vars['version'] == '4.5.6'
+
+    # check selectors
+    assert cm.meta['source']['sha256__###conda-selector###__py2k'] == 1
+    assert cm.meta['source']['sha256__###conda-selector###__py3k and win'] == 5
+
+    # check other keys
+    assert cm.meta['build']['number'] == 10
+    assert cm.meta['package']['name'] == '{{ name|lower }}'
+    assert cm.meta['source']['url'] == 'foobar'
+
```

Comment:
I'd add a round trip test here as well

Reply:
ruamel.yaml cleans lint so I had to added a linted version of the input.